src/entities/createActionListenerMiddleware.test.ts

```diff
@@ -0,0 +1,236 @@
+import { configureStore } from '../configureStore'
+import {
+  createActionListenerMiddleware,
+  addListenerAction,
+  removeListenerAction
+} from './createActionListenerMiddleware'
+import { createAction } from '../createAction'
+
+const middlewareApi = {
+  getState: expect.any(Function),
+  dispatch: expect.any(Function)
+}
+
+describe('createActionListenerMiddleware', () => {
+  let store = configureStore({
+    reducer: () => ({}),
+    middleware: [createActionListenerMiddleware()] as const
+  })
+  let reducer: jest.Mock
+  let middleware: ReturnType<typeof createActionListenerMiddleware>
+
+  const testAction1 = createAction<string>('testAction1')
+  type TestAction1 = ReturnType<typeof testAction1>
+  const testAction2 = createAction<string>('testAction2')
+
+  beforeEach(() => {
+    middleware = createActionListenerMiddleware()
+    reducer = jest.fn(() => ({}))
+    store = configureStore({
+      reducer,
+      middleware: [middleware] as const
+    })
+  })
+
+  test('directly subscribing', () => {
+    const listener = jest.fn((_: TestAction1) => {})
+
+    middleware.addListener(testAction1, listener)
+
+    store.dispatch(testAction1('a'))
+    store.dispatch(testAction2('b'))
+    store.dispatch(testAction1('c'))
+
+    expect(listener.mock.calls).toEqual([
+      [testAction1('a'), middlewareApi],
+      [testAction1('c'), middlewareApi]
+    ])
+  })
+
+  test('subscribing with the same listener will not make it trigger twice (like EventTarget.addEventListener())', () => {
+    /**
+     * thoughts: allow to use this to override the options for a listener?
+     * right now it's just exiting if the listener is already registered
+     */
+
```

Comment:
Any thoughts on that?

Reply:
Don't feel like there's a lot of reason to change the options for a listener once added.

Reply:
Yeah, one could always unsubscribe & resubscribe with different options. No need to make it too complicated.